### PR TITLE
Add ALE

### PR DESCRIPTION
**Pull request by j0hnmeow**
ALE is a Vim/Neovim plugin providing LSP client and linting with a big number of linters

### CODE REVIEW

**Review by cooperlees — APPROVED** [review]
LGTM and a simple alternative if the official way is not for the user ...

**Review by ichard26 — APPROVED** [review]
Many thanks!

**Comment by ichard26** [comment]
@j0hnmeow could you resolve the merge conflict and @cooperlees or @JelleZijlstra could one of you merge?

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
I just fixed the conflict. I'll merge once CI is (mostly) green.

File: docs/editor_integration.md [hunk 1]
```diff
@@ -233,6 +235,16 @@ If you later want to update _Black_, you should do it like this:
...
+### With ALE
+
+1. Install [`ale`](https://github.com/dense-analysis/ale)
+2. Install `black`
+3. Add this to your vimrc:
+
+   ```vim
+   let g:ale_fixers = {}
+   let g:ale_fixers.python = ['black']
+   ```
```
Comment (ichard26) [inline docs/editor_integration.md:247]: *sigh*, this still fails the lint status check. I would fix it myself, but I can't... @JelleZijlstra can you add an extra line after this end of code fence?

Comment (JelleZijlstra) [reply docs/editor_integration.md:247]: done

**Pull request merged**